src/lib/scene-internal.ts

```diff
@@ -713,7 +716,6 @@ export class SceneInternal implements Renderer {
    * the canvas to convert it to a blob.
    */
   async snapshot(): Promise<Blob> {
-    this.drawCommand();
```

Comment:
What's this for? (Just making sure the deletion is intentional)

Reply:
Yes, that was intentional. Previously, when `preserveDrawingBuffer` was `false`, it was prudent to perform the draw immediately before reading back the blob to ensure that the data was available to read. Now that `preserveDrawingBuffer` is `true`, the previous draw is sufficient. Tests pass with this change.

Currently, the `snapshot()` method is not exposed to API users. It is only used by tests to perform comparisons on rendered pixels.